src/caffe/common.cpp

```diff
@@ -192,6 +201,26 @@ void Caffe::DeviceQuery() {
   return;
 }
 
+bool Caffe::CheckDevice(const int device_id) {
+  bool r =  ((cudaSuccess == cudaSetDevice(device_id)) &&
+            (cudaSuccess == cudaFree(0)));
+  cudaGetLastError();
+  return r;
+}
+
+int Caffe::GrabDevice(const int start_id) {
```

Comment:
Would `FindDevice` be a better name here? To me "grab" implies that this method is claiming the device, but it's merely a passive check.